x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -0,0 +1,168 @@
+                  await enableRule({ rule, alertsClient, savedObjectsClient });
+                }
+              })
+            );
+            break;
+          case BulkAction.disable:
+            await Promise.all(
+              rules.data.map(async (rule) => {
+                if (rule.enabled) {
+                  throwHttpError(await mlAuthz.validateRuleType(rule.params.type));
+                  await alertsClient.disable({ id: rule.id });
+                }
+              })
+            );
+            break;
+          case BulkAction.delete:
+            await Promise.all(
+              rules.data.map(async (rule) => {
+                const ruleStatuses = await ruleStatusClient.find({
+                  perPage: 6,
+                  search: rule.id,
+                  searchFields: ['alertId'],
+                });
+                await deleteRules({
+                  alertsClient,
+                  savedObjectsClient,
+                  ruleStatusClient,
+                  ruleStatuses,
+                  id: rule.id,
+                });
+              })
+            );
+            break;
+          case BulkAction.duplicate:
+            await Promise.all(
+              rules.data.map(async (rule) => {
+                throwHttpError(await mlAuthz.validateRuleType(rule.params.type));
+
+                const createdRule = await alertsClient.create({
+                  data: duplicateRule(rule),
+                });
+
+                const ruleActions = await getRuleActionsSavedObject({
+                  savedObjectsClient,
+                  ruleAlertId: rule.id,
+                });
+
+                await updateRulesNotifications({
+                  ruleAlertId: createdRule.id,
+                  alertsClient,
+                  savedObjectsClient,
+                  enabled: createdRule.enabled,
+                  actions: ruleActions?.actions || [],
+                  throttle: ruleActions?.alertThrottle,
+                  name: createdRule.name,
+                });
+              })
+            );
+            break;
+          case BulkAction.export:
+            const exported = await getExportByObjectIds(
+              alertsClient,
+              rules.data.map(({ params }) => ({ rule_id: params.ruleId }))
+            );
+
+            const responseBody = `${exported.rulesNdjson}${exported.exportDetails}`;
+
+            return response.ok({
+              headers: {
+                'Content-Disposition': `attachment; filename="rules_export.ndjson"`,
+                'Content-Type': 'application/ndjson',
+              },
+              body: responseBody,
+            });
+        }
+
+        return response.ok({ body: { success: true, rules_count: rules.data.length } });
+      } catch (err) {
+        const error = transformError(err);
+        return siemResponse.error({
+          body: error.message,
+          statusCode: error.statusCode,
+        });
+      }
+    }
+  );
+};
```

Comment:
Do we have any e2e for this one? I don't think I saw one? We typically try to have a e2e for each route we add. Examples are here:
https://github.com/elastic/kibana/blob/master/x-pack/test/detection_engine_api_integration/security_and_spaces/tests/export_rules.ts

Reply:
Added some e2e tests: https://github.com/elastic/kibana/pull/100554/files#diff-033a39080b5edd054aecd3ddfa4a06fc9d7e145e71bda681a8bd3647efb21956
Thanks for noticing 👍